apps/src/lib/levelbuilder/lesson-editor/resourcesEditorRedux.js

```diff
@@ -29,6 +35,13 @@ export default function resources(state = [], action) {
       newState = newState.concat([action.newResource]);
       break;
     }
+    case EDIT_RESOURCE: {
+      const resourceToEdit = newState.find(
+        resource => resource.key === action.updatedResource.key
+      );
+      Object.assign(resourceToEdit, action.updatedResource);
+      break;
+    }
```

Comment:
if helpful, since your input is a resourceShape, you could explicitly check the type here, like this: https://github.com/code-dot-org/code-dot-org/blob/398cad85c1500ff2ba3fb9e33489752a1b46ae03/apps/src/lib/levelbuilder/lesson-editor/activitiesEditorRedux.js#L627-L632

IIRC, any unit tests causing type mismatches would fail.